src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,82 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+
+import { User } from '../../types/User';
+
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import {
+  clearCurrentTodo,
+  selectCurrentTodo,
+} from '../../features/currentTodo';
+
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+export const TodoModal: FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<null | User>(null);
+
+  const currentTodo = useAppSelector(selectCurrentTodo);
+  const dispatch = useAppDispatch();
+
+  const handleCloseModal = () => {
+    dispatch(clearCurrentTodo());
+  };
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  }, [currentTodo]);
```

Comment:
Consider adding a condition to reset the loading state and clear the user data if `currentTodo` changes to `null`. This will ensure the component's state is consistent when there is no current todo.